sn_cli/src/files/files_uploader.rs

```diff
@@ -0,0 +1,337 @@
+// Copyright 2024 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+use crate::ChunkManager;
+use bytes::Bytes;
+use color_eyre::{eyre::eyre, Report, Result};
+use futures::StreamExt;
+use rand::prelude::SliceRandom;
+use rand::thread_rng;
+use sn_client::{
+    transfers::{TransferError, WalletError},
+    Client, Error as ClientError, UploadCfg, UploadEvent, UploadSummary, Uploader,
+};
+use sn_protocol::storage::Chunk;
+use std::{
+    path::{Path, PathBuf},
+    time::{Duration, Instant},
+};
+use tokio::{sync::mpsc::Receiver, task::JoinHandle};
+use tracing::{debug, error, info, warn};
+use walkdir::{DirEntry, WalkDir};
+use xor_name::XorName;
+
+use super::get_progress_bar;
+
+/// Combines the `Uploader` along with the `ChunkManager`
+pub struct FilesUploader {
+    client: Client,
+    root_dir: PathBuf,
+
+    // entries to upload
+    entries_to_upload: Vec<DirEntry>,
+    // todo: remove this in favour of an trait that abstracts the std out printing.
+    file_paths_to_print: Vec<PathBuf>,
+
+    // config
+    make_data_public: bool,
+    upload_cfg: UploadCfg,
+}
+
+impl FilesUploader {
+    pub fn new(client: Client, root_dir: PathBuf) -> Self {
+        Self {
+            client,
+            root_dir,
+
+            entries_to_upload: Default::default(),
+            file_paths_to_print: Default::default(),
+
+            make_data_public: false,
+            upload_cfg: Default::default(),
+        }
+    }
+
+    pub fn set_upload_cfg(mut self, cfg: UploadCfg) -> Self {
```

Comment:
I've looked into this a little bit,. but are getters and setters necessary in Rust?

I know these are cherry picked but I see it mentioned now again:
https://users.rust-lang.org/t/best-practices-for-visibility-of-fields/56087/3
https://users.rust-lang.org/t/best-practices-on-setters-and-accessor-methods-in-general/66530/2

Reply:
(just a consideration, no worries here)